pom.xml

```diff
@@ -238,6 +238,17 @@
             <artifactId>juniversalchardet</artifactId>
             <version>2.4.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.30</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
```

Comment:
Did we also consider Google's Guava? (Just wondering, not a blocker by any means)

Nice to see Lombok, although I believe it hinders a bit bc it becomes heavily IDE-dependent

Reply:
Also: We are only using two methods from `commons-collections` it seems. Can't we perhaps inline? TBH, it looks like an Idiom we could evolve further into the `Util` class wdyt?

Reply:
lombok-wise, why not leverage commons-lang`s builders classes instead?

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html